src/components/Header/Button.tsx

```diff
@@ -1,10 +1,20 @@
-import { React } from 'react'
+import React, { FC } from 'react'
 import { css } from '@emotion/core'
-import { Link } from 'gatsby'
+import { GatsbyLinkProps, Link } from 'gatsby'
 import { useTheme } from '../Theming'
 import { rgba, darken } from 'polished'
 
-const Button = ({ to, children, secondary, ...restProps }) => {
+type LinkPropsType = Omit<GatsbyLinkProps<unknown>, 'ref'> & {
+  secondary?: boolean;
+};
+
+type HTMLButtonElementPropsType = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  secondary?: boolean;
+};
+
+type ButtonPropsType = LinkPropsType | HTMLButtonElementPropsType;
+
+const Button: FC<ButtonPropsType> = ({ children, secondary, ...restProps }) => {
```

Comment:
1. I put `to` to `restProps`
2. Below I use `as LinkPropsType` to check if `to` exists, otherwise cast it to `as HTMLButtonElementPropsType` (didn't find any better way)